src/commands.rs

```diff
@@ -4009,11 +4030,18 @@ fn is_fast_forward(repo: RepoRef, branch_name: &str, new_target_id: &CommitId) -
     }
 }
 
-fn cmd_branch(ui: &mut Ui, command: &CommandHelper, args: &BranchArgs) -> Result<(), CommandError> {
+fn cmd_branch(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &BranchSubcommand,
+) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let branch_names: Vec<&str> = if args.delete || args.forget {
-        let view = workspace_command.repo().view();
-        args.names
+    let view = workspace_command.repo().view();
+    fn assert_branch_names_exist<'a>(
```

Comment:
For me, the name `assert` is so strongly associated with checks for programming errors (typically using some language-provided `assert` keyword). How about `check` or `validate` instead?